src/preferences/dialog/dlgprefwaveform.h

```diff
@@ -26,6 +30,12 @@ class DlgPrefWaveform : public DlgPreferencePage, public Ui::DlgPrefWaveformDlg
   private slots:
     void slotSetFrameRate(int frameRate);
     void slotSetWaveformType(int index);
+    void slotSetWaveformEnabled(bool checked);
+    void slotSetWaveformAcceleration(bool checked);
+#ifdef MIXXX_USE_QOPENGL
+    DECL_SLOT_WAVEFORM_OPTION(SplitStereoSignal);
+    DECL_SLOT_WAVEFORM_OPTION(HighDetails);
+#endif
```

Comment:
I don't think it makes sense to generate these here. It would be better to have a single 
```cpp
slotSetWaveformOptions(allshader::WaveformRendererSignalBase opts, checked) {
    // pseudocode
    if (checked) {
        m_options |= opts;
    } else {
        m_options &= !opts;
    }
}
```

Then use the functor `connect` calls to partially bind parameters in the lambda ([avoid `std::bind`](https://abseil.io/tips/108))

```cpp
connect(..., [this](bool checked){slotSetWaveformOptions(SplitStereoSignal, checked)}
```

Reply:
What is the benefit from that approach? I tend to dislike the use of lambda-connect when possible as they usually make things more opaque IMHO. For me, atomic slots are easier to read than generic one with a lambda in between.

Reply:
it avoids the macro and it makes the code more readable because I can grep for the function if I need to. Otherwise I'd need to rely on my editors "jump to definition" functionality. ["Official Guidance"](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es30-dont-use-macros-for-program-text-manipulation) also discourages this style because its a bad tradeoff. 

Can you elaborate a bit on what you mean with atomic vs generic slots?

Reply:
What I meant by atomic is the fact that Qt will connect the signal and the slot with intermediary transformation. 
```
connect(checkbox, QCheckBox::toggled, mywaveform, &Pref::slotSetWaveformAcceleration);
```
This way you don't need to look at the function itself as is performing a single atomic operation, which is flip a bit.

> ["Official Guidance"](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es30-dont-use-macros-for-program-text-manipulation) also discourages this style because its a bad tradeoff.

I'm slightly confused about that since:
1. Qt uses macros everywhere.
2. We've introduces macros with some of the latest QML feature that were merged.

Reply:
> This way you don't need to look at the function itself as is performing a single atomic operation, which is flip a bit.

I don't think thats apparent or given from the name. It also applies to a hypothetical

```cpp
connect(checkbox, QCheckBox::toggled, mywaveform, [](bool checked) {
    slotSetWaveformAcceleration(checked);
});
```
Anyways. If you don't like lambdas used here, wdyt about the following compromise: Still implement a single `setWaveformOptions` that essentially contains the code thats currently in the macro. Introduce `slotSetWaveformSplitStereoSignal` as a member function, that just does `slotSetWaveformOptions(SplitStereoSignal, checked)`? The `QObject::connect` call stays the same, in the readable style that you prefer and we still avoid the macro. After all we've had two distinct discussions until now, `connect` with lambda and avoiding the macro, which are basically orthogonal.

> Qt uses macros everywhere.

Yeah, unfortunately, because Qt is Qt and bound to the APIs they came up 25 years ago. 

> We've introduces macros with some of the latest QML feature that were merged.

Yes, in that case the tradeoff was clearly worth it since it avoided boilerplate that was otherwise not possible to avoid. In this case there is almost no extra boilerplate, so the tradeoff tips to the other side.

Reply:
Fair enough, I'll make the suggested change